src/so3/4C_so3_surface_evaluate.cpp

```diff
@@ -212,6 +208,7 @@ int Discret::Elements::StructuralSurface::evaluate_neumann(Teuchos::ParameterLis
   {
     // --------------------------------------------------
     // get knotvector
+    auto* nurbsdis = dynamic_cast<Core::FE::Nurbs::NurbsDiscretization*>(&(discretization));
```

Comment:
why do you cast to a pointer and not to a reference?
```suggestion
    auto& nurbsdis = dynamic_cast<Core::FE::Nurbs::NurbsDiscretization>(discretization);
```

Reply:
I think what @amgebauer wants to say is: you can now cast to a reference because you know that the cast will work.

Reply:
yes. There is no further check anyway.

Reply:
Yes, you're both right. I just applied the dynamic_cast that was previously in the code.